### PR TITLE
SubNav tweaks so that the most specific section (or tag) is the one that is highlighted

### PR DESCRIPTION
**Pull request by NataliaLKB**
## What does this change?
For the new nav, we have a subnav:
image

At the moment, a context word should be highlighted for each article and front on both the subNav and the Nav 5 primary words. This is only happening for sections which is a problem, because the context word isn't accurate and actually could change unexpectedly for the user.

Current journey:
broken-subnav-journey

Proposed journey:
better-subnav-journey

## What is the value of this and can you measure success?
Nicer user experience!

## Does this affect other platforms - Amp, Apps, etc?
No

## Screenshots
image

## Tested in CODE?
Yes

### CODE REVIEW

**Comment by NataliaLKB** [comment]
cc @guardian/dotcom-platform

File: common/app/common/NewNavigation.scala [hunk 1]
```diff
@@ -454,6 +481,21 @@ object NewNavigation {
       editionalisedSubSectionLinks.exists(_.pageId == sectionId)
     }
 
+    def getSectionOrTagId(page: Page) = {
+      val tags = Navigation.getTagsFromPage(page)
+      val intersectionalKeywords = tagPages.intersect(tags.keywordIds)
```
Comment (jfsoul) [inline common/app/common/NewNavigation.scala]: what about renaming this to `commonKeywords` or something?

**Review by jfsoul — APPROVED** [review]
Nice work, looks good to me!

File: common/app/common/NewNavigation.scala (hunk 1 above)
Comment (NataliaLKB) [inline common/app/common/NewNavigation.scala]: Haha yes that is a much less jargon-y name

**Comment by paperboyo** [comment]
Superb! Are we planing to do anything about the repetitions?

repetition

**Comment by NataliaLKB** [comment]
@paperboyo that is something that has been talked about by @stephanfowler and Alex B. But this change needs to go live before we change fronts.

Definitely something we want to do though!

**Comment by paperboyo** [comment]
We should also find a way to represent third (and further) level(s) of hierarchy. Nesting is more obvious in the old nav, I'm afraid:

nav confusion

vs.

picture 96

**Pull request merged**